src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,426 @@
+"""
+Algorithms for the camera calibration extraction.
+"""
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy import units as u
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import DL1CameraContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    Int,
+    List,
+    TelescopeParameter,
+    IntTelescopeParameter,
+    ComponentName,
+)
+from ctapipe.image.extractor import ImageExtractor
+
+__all__ = ["CalibrationExtractor", "FlatFieldExtractor", "PedestalExtractor"]
+
+
+class CalibrationExtractor(TelescopeComponent):
+    """
+    Base component for camera calibration extractors.
+    Fills the MonitoringCameraContainer on the base of a given sample of interleaved
+    calibration events.
+    The sample is defined by a minimal number of calibration events (sample_size) and
+    the statistics are calculated at a given frequency (update_frequency).
+    The extractors are supposed to be called in an event loop, extract (image_extractor_type)
+    and collect the event charge and arrival time. At a pre-defined iteration frequency, the
+    sample statistics are calculated and the corresponding MonitoringCameraContainer is filled.
+
+    The pixels are set as outliers on the base of a cut on the pixel charge median over the full
+    sample distribution (charge_median_cut_outliers) and the pixel charge standard deviation over
+    the full sample distribution with respect to the camera median values (charge_std_cut_outliers),
+    as well as the pixel signal/arrival time inside the waveform time (time_cut_outliers). Sigma
+    clipping from astropy is applied for the outlier removal using configurable parameters
+    (sigma_clipping_max_sigma and sigma_clipping_iterations).
+    # TODO:
+        1) Discuss whether to express the outlier cuts based on quantiles.
+        2) Discuss whether to calculate the number of iterations for sigma clipping on the fly.
+
+    Attributes
+    ----------
+    sample_size : int
+        Minimal number of calibration events requested for the statistics
+    update_frequency : int
+        Calculation frequency of the statistics
```

Comment:
This value is not clear to me from the explanation given.

Reply:
I agree, this is not clear